### PR TITLE
Revive the project [v1.0]

### PR DESCRIPTION
**Pull request by Exelord**
This PR will introduce:

- [x] Fixed cannot helper
- [x] Upgraded to Ember 3.0
- [x] Limited support to 2.12 and up.
- [x] Refactored code and migrated tests to latest API
- [x] Improved performance

## Todo

- [ ] Add more tests
- [ ] Update readme
- [ ] Add code documentation

## Notable changes

The public method of `can` service - `build` has changed his API. Now it receives already parsed `abilityName`.

#### Usage Before
```js
let ability = this.build(abilityString, model, properties);
```

#### Usage After
```js
let { abilityName } = this.parse(abilityString);
let ability = this.build(abilityName, model, properties);
```

### CODE REVIEW

**Comment by Exelord** [comment]
@sukima can you review this?

**Comment by sukima** [comment]
@Exelord I'm at EmberConf this week. Will look at this as soon as I have a free moment.

**Comment by Exelord** [comment]
Everyone is on EmberConf -.- ... except me 🤕.
Don't worry :D We ve got time.

**Comment by knownasilya** [comment]
Any chance this can get released?

File: addon/initializers/setup-ember-can.js [hunk 1]
```diff
@@ -0,0 +1,11 @@
+import Resolver from 'ember-resolver';
+
+Resolver.reopen({
+  // eslint-disable-next-line ember/avoid-leaking-state-in-ember-objects
```
Comment (sukima) [inline addon/initializers/setup-ember-can.js]: What makes this line leaking I wonder.

**Comment by sukima** [comment]
@Exelord / @knownasilya Sorry this took so long to get to. I skimmed most of this and I think it is pretty Good To Go!. Tests pass. Great upgrades. All new styles. Yeah, :+1: :shipit:

File: addon/initializers/setup-ember-can.js (hunk 1 above)
Comment (Exelord) [reply addon/initializers/setup-ember-can.js]: Using {} or [] ... but in this case we dont have too much choice :/

**Comment by Exelord** [comment]
@rlivsey @blimmer Is there any chance to have it or I just waste few hours?

**Comment by blimmer** [comment]
@Exelord as I mentioned before I don't have the time to help out a ton with this project and it fell of my radar. Please understand that we're all trying to do our best with the resources and time we have.

@rlivsey can you please get @Exelord the ability to manage this project? It seems like he has the bandwidth to help maintain this project.

**Comment by blimmer** [comment]
I didn't review but it has a checkmark already - merging.

**Comment by Exelord** [comment]
@blimmer I fully understand your situation! We don't expect to have releases every week/month but it would be good to have min 1 active person for publishing the project. I think the community will take a responsibility for the rest if we give a feeling that this project is not dead.

Thanks for moving with this one! :)

**Pull request merged**